### PR TITLE
feat: add Spanner samples for MR CMEK

### PR DESCRIPTION
**Pull request by panerorenn9541**
Add samples and tests for create database, create backup, copy backup, and restore backup.

### CODE REVIEW

File: spanner/api/Spanner.Samples.Tests/CreateBackupWithMRCMEKAsyncTest.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        Skip.If(!_fixture.RunCmekBackupSampleTests, SpannerFixture.SkipCmekBackupSamplesMessage);
+        // Create a backup with custom encryption keys.
+        var sample = new CreateBackupWithMRCMEKAsyncSample();
+        var backup = await sample.CreateBackupWithMRCMEKAsync(_fixture.ProjectId, _fixture.InstanceId, _fixture.FixedEncryptedDatabaseId, _fixture.EncryptedBackupId, _fixture.KmsKeyNames);
+        Assert.Equal(_fixture.KmsKeyNames.Length, backup.EncryptionInfo.KmsKeyVersionsAsCryptoKeyVersionNames.Length);
+        foreach (CryptoKeyVersionName KmsKeyVersion in backup.EncryptionInfo.KmsKeyVersionsAsCryptoKeyVersionNames)
+        {
+          Assert.True(_fixture.KmsKeyNames.contains(KmsKeyVersion.CryptoKeyId));
+        }
+    }
```
Comment (amanda-tarafa) [inline spanner/api/Spanner.Samples.Tests/CreateBackupWithMRCMEKAsyncTest.cs:44]: Probably cleaner to do something like:

```suggestion
        Assert.All(backup.EncryptionInfo.KmsKeyVersionsAsCryptoKeyVersionNames, keyName => _fixture.KmsKeyNames.Contains(keyName.CriptoKeyId));
```

File: spanner/api/Spanner.Samples.Tests/CreateDatabaseWithMRCMEKAsyncTest.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    {
+        _fixture = fixture;
+    }
+
+    [Fact]
+    public async Task TestCreateDatabaseWithMRCMEKAsync()
+    {
+        // Create a database with custom encryption keys.
+        var sample = new CreateDatabaseWithMRCMEKAsyncSample();
+        var database = await sample.CreateDatabaseWithMRCMEKAsync(_fixture.ProjectId, _fixture.InstanceId, _fixture.EncryptedDatabaseId, _fixture.KmsKeyNames);
```
Comment (amanda-tarafa) [inline spanner/api/Spanner.Samples.Tests/CreateDatabaseWithMRCMEKAsyncTest.cs:37]: Similarly, use a different database ID here, this one is used for the plain encryption configuration.

File: spanner/api/Spanner.Samples/CopyBackupWithMRCMEK.cs [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+            EncryptionConfig = new CopyBackupEncryptionConfig
+            {
+                EncryptionType = CopyBackupEncryptionConfig.Types.EncryptionType.CustomerManagedEncryption,
+                KmsKeyNamesAsCryptoKeyNames = kmsKeyNames,
+            }
+        };
+
+        var response = databaseAdminClient.CopyBackup(request);
+        Console.WriteLine("Waiting for the operation to finish.");
+        var completedResponse = response.PollUntilCompleted(new PollSettings(Expiration.FromTimeout(TimeSpan.FromMinutes(15)), TimeSpan.FromMinutes(2)));
```
Comment (amanda-tarafa) [inline spanner/api/Spanner.Samples/CopyBackupWithMRCMEK.cs:46]: I'm guessing default poll settings are not enough here?

File: spanner/api/Spanner.Samples/CopyBackupWithMRCMEK.cs [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        {
+            Console.WriteLine($"Error while creating backup: {completedResponse.Exception}");
+            throw completedResponse.Exception;
+        }
+
+        Backup backup = completedResponse.Result;
+
+        Console.WriteLine($"Backup created successfully.");
+        Console.WriteLine($"Backup with Id {sourceBackupId} has been copied from {sourceProjectId}/{sourceInstanceId} to {targetProjectId}/{targetInstanceId} Backup {targetBackupId}");
+        Console.WriteLine($"Backup {backup.Name} of size {backup.SizeBytes} bytes was created with encryption keys {0} at {backup.CreateTime} from {backup.Database} and is in state {backup.State} and has version time {backup.VersionTime.ToDateTime()}", string.Join(", ", kmsKeyNames));
```
Comment (amanda-tarafa) [inline spanner/api/Spanner.Samples/CopyBackupWithMRCMEK.cs:58]: Here and everywhere else where this is present:
```suggestion
        Console.WriteLine($"Backup {backup.Name} of size {backup.SizeBytes} bytes was created with encryption keys {string.Join(", ", kmsKeyNames)} at {backup.CreateTime} from {backup.Database} and is in state {backup.State} and has version time {backup.VersionTime.ToDateTime()}" );
```

File: spanner/api/Spanner.Samples.Tests/CopyBackupWithMRCMEKTest.cs [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+    public CopyBackupWithMRCMEKTest(SpannerFixture spannerFixture)
+    {
+        _spannerFixture = spannerFixture;
+    }
+
+    [Fact]
+    public void CopyBackupWithMRCMEK()
+    {
+        CopyBackupWithMRCMEKSample copyBackupWithMRCMEKSample = new CopyBackupWithMRCMEKSample();
```
Comment (amanda-tarafa) [inline spanner/api/Spanner.Samples.Tests/CopyBackupWithMRCMEKTest.cs:32]: You cannot use the same backup IDs that are being used for the other backup tests. You need specific IDs for these tests, and make certain the backups are there when you need them, etc.

**Review by amanda-tarafa — CHANGES_REQUESTED** [review]
Issues I've commented on appear on several places even though I flag them on a single place, please fix all ocurrences.

This code doesn't build, see the [build log](https://btx.cloud.google.com/invocations/e5447f07-afb7-4f48-807d-c68bab339d39/log) for errors.

When you have run the tests locally and they pass, let me know and I can run them on the PR with the backup tests enabled to make certain they run on CI before merging. At the moment these new tests will conflict with the existing backup samples and tests.

File: spanner/api/Spanner.Samples/CopyBackupWithMRCMEK.cs (hunk 3 above)
Comment (panerorenn9541) [reply spanner/api/Spanner.Samples/CopyBackupWithMRCMEK.cs]: Changed to use default poll settings

File: spanner/api/Spanner.Samples.Tests/SpannerFixture.cs [hunk 6]
```diff
@@ -121,9 +143,13 @@ public async Task InitializeAsync()
 
         // Create encryption key for creating an encrypted database and optionally backing up and restoring an encrypted database.
         await InitializeEncryptionKeys();
+        KmsKeyNames.Append(KmsKeyName);
+        KmsKeyNames.Append(KmsKeyName2);
+        KmsKeyNames.Append(KmsKeyName3);
```
Comment (amanda-tarafa) [inline spanner/api/Spanner.Samples.Tests/SpannerFixture.cs:148]:
```suggestion
       
```

File: spanner/api/Spanner.Samples.Tests/SpannerFixture.cs [hunk 7]
```diff
@@ -72,11 +72,31 @@ AlbumTitle STRING(MAX)
...
+        Environment.GetEnvironmentVariable("spanner.test.key.project") ?? Environment.GetEnvironmentVariable("GOOGLE_PROJECT_ID"),
+        Environment.GetEnvironmentVariable("spanner.test.key.location2") ?? "us-east1",
+        Environment.GetEnvironmentVariable("spanner.test.key.ring2") ?? "spanner-test-keyring2",
+        Environment.GetEnvironmentVariable("spanner.test.key.name2") ?? "spanner-test-key2");
+    public CryptoKeyName KmsKeyName3 { get; } = new CryptoKeyName(
+        Environment.GetEnvironmentVariable("spanner.test.key.project") ?? Environment.GetEnvironmentVariable("GOOGLE_PROJECT_ID"),
+        Environment.GetEnvironmentVariable("spanner.test.key.location3") ?? "us-east4",
+        Environment.GetEnvironmentVariable("spanner.test.key.ring3") ?? "spanner-test-keyring3",
+        Environment.GetEnvironmentVariable("spanner.test.key.name3") ?? "spanner-test-key3");
+    public CryptoKeyName[] KmsKeyNames { get; private set; }
```
Comment (amanda-tarafa) [inline spanner/api/Spanner.Samples.Tests/SpannerFixture.cs:99]:
```suggestion
    public CryptoKeyName[] KmsKeyNames { get; } = new CryptoKeyName[] { KmsKeyName, KmsKeyName1, KmsKeyName2 };

```

**Review by amanda-tarafa — CHANGES_REQUESTED** [review]

**Comment by amanda-tarafa** [comment]
Closing in favour of #2832 . These are very similar, changea are mostly renaming, and also in how we use some fixture elements. Sample code is identical.

**Pull request closed without merging**